### ISSUE TITLE
Add MacOS (Darwin) Support

### ISSUE THREAD

**Issue by dblock** [opensearch-project/opensearch-plugins#96]
**Is your feature request related to a problem?**

Coming from https://github.com/opensearch-project/opensearch-build/issues/38. In order to release a MacOS distribution each plugin needs to work on MacOS.

**What solution would you like?**

Every component has passing CI on MacOS (`macos-latest` on GHA) and has MacOS-specific documentation, if needed.

- [ ] https://github.com/opensearch-project/alerting-dashboards-plugin/issues/138
- [x] https://github.com/opensearch-project/alerting/issues/215
- [ ] https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/issues/105
- [x] https://github.com/opensearch-project/anomaly-detection/issues/275
- [x] https://github.com/opensearch-project/asynchronous-search/issues/52
- [x] https://github.com/opensearch-project/dashboards-notebooks/issues/92
- [x] https://github.com/opensearch-project/dashboards-reports/issues/192
- [x] https://github.com/opensearch-project/dashboards-visualizations/issues/34
- [x] https://github.com/opensearch-project/index-management-dashboards-plugin/issues/115
- [x] https://github.com/opensearch-project/index-management/issues/173
- [x] https://github.com/opensearch-project/job-scheduler/issues/76
- [ ] https://github.com/opensearch-project/k-NN/issues/158
- [ ] https://github.com/opensearch-project/notifications/issues/359
- [ ] https://github.com/opensearch-project/performance-analyzer/issues/76
- [ ] https://github.com/opensearch-project/security-dashboards-plugin/issues/846
- [ ] https://github.com/opensearch-project/security/issues/1475
- [x] https://github.com/opensearch-project/sql/issues/250
- [ ] https://github.com/opensearch-project/trace-analytics/issues/161
- [x] https://github.com/opensearch-project/cross-cluster-replication/issues/219

**Comment by dblock** [comment]
> Should we finish this in 1.2 release?

It's in [the roadmap under 1.3](https://github.com/orgs/opensearch-project/projects/1) currently, but don't wait on other plugins to add MacOS CI :)

**Comment by dbbaughe** [comment]
@dblock I'm assuming plugins are blocked from moving forward on this until core publishes the min mac distribution?

**Comment by dblock** [comment]
> @dblock I'm assuming plugins are blocked from moving forward on this until core publishes the min mac distribution?

Plugins should be able to build and CI on MacOS without core. Maven dependencies aren't platform-specific.

**Comment by dbbaughe** [comment]
> > @dblock I'm assuming plugins are blocked from moving forward on this until core publishes the min mac distribution?
> 
> Plugins should be able to build and CI on MacOS without core. Maven dependencies aren't platform-specific.

Wouldn't that mean macos development is solved then and all plugin developers should be able to develop on their local macos? Has that been fixed? I know we have a hacky workaround to switch from ARCHIVE to INTEG_TEST for the distribution, but that does not include any of the extra modules which means a lot of plugin functionality is broken.

**Comment by dblock** [comment]
> > > @dblock I'm assuming plugins are blocked from moving forward on this until core publishes the min mac distribution?
> > 
> > 
> > Plugins should be able to build and CI on MacOS without core. Maven dependencies aren't platform-specific.
> 
> Wouldn't that mean macos development is solved then and all plugin developers should be able to develop on their local macos? Has that been fixed? I know we have a hacky workaround to switch from ARCHIVE to INTEG_TEST for the distribution, but that does not include any of the extra modules which means a lot of plugin functionality is broken.

Yes, that too. I don't know anything about ARCHIVE vs. INTEG_TEST, point to an issue? We should fix it.

**Comment by derek-ho** [comment]
@dblock is this still planned, or should we close this out?

**Comment by dblock** [comment]
This is still valid.
